src/assisted_test_infra/test_infra/utils/logs_utils.py

```diff
@@ -50,15 +133,15 @@ def verify_logs_uploaded(
                 f"{tar.getnames()} " f"logs are less than minimum of {expected_min_log_num}"
             )
             tar.extractall(tempdir)
-            for gz in os.listdir(tempdir):
-                if "bootstrap" in gz and verify_bootstrap_errors is True:
-                    _verify_node_logs_uploaded(tempdir, gz)
-                    _verify_bootstrap_logs_uploaded(tempdir, gz, installation_success, verify_control_plane)
-                elif "master" in gz or "worker" in gz:
-                    _verify_node_logs_uploaded(tempdir, gz)
-                elif "controller" in gz:
-                    if check_oc:
-                        _verify_oc_logs_uploaded(os.path.join(tempdir, gz))
+            _verify_duplicate_and_size_logs_file(tempdir)
```

Comment:
shouldn't this function be called in every directories?

Reply:
Checking

Reply:
When we call to this function here we accept
` ['test-infra-cluster-6c2d0ab7_master_test-infra-cluster-6c2d0ab7-master-2.tar', 'test-infra-cluster-6c2d0ab7_master_test-infra-cluster-6c2d0ab7-master-0.tar', 'test-infra-cluster-6c2d0ab7_bootstrap_test-infra-cluster-6c2d0ab7-master-1.tar', 'cluster_events.json', 'cluster_metadata.json', 'controller_logs.tar.gz']`

so we have only files tar that we can check base on the files that we dont have any duplicate content by md5 plus duplicate subset naming of files. we also here verify that tar files are not huge as we had in the bug you saw above (duplicate files + huge files)

Reply:
Now for the deeper inspection inside the files , it depends on the stage and if its boot strap and it should be handled in the called function ->  line 138 if else section . i know its not optimal and there is a better way ... i want to at least have a test for the positive case that we install successfully and no issues .... 

I think this is the first patch for common usage and i will have to add mores ..... some results includes file and other do not ... so trying on the safe side

Reply:
To avoid any confusion, can you just add a comment in the code stating what just said? (That basically, there only one level of files)